python/cudf_polars/cudf_polars/experimental/rapidsmpf/join.py

```diff
@@ -0,0 +1,242 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Join logic for the RapidsMPF streaming runtime."""
+
+from __future__ import annotations
+
+import asyncio
+from typing import TYPE_CHECKING, Any, Literal
+
+from rapidsmpf.streaming.core.message import Message
+from rapidsmpf.streaming.cudf.table_chunk import TableChunk
+
+import pylibcudf as plc
+
+from cudf_polars.containers import DataFrame
+from cudf_polars.dsl.ir import IR, Join
+from cudf_polars.experimental.rapidsmpf.dispatch import (
+    generate_ir_sub_network,
+)
+from cudf_polars.experimental.rapidsmpf.nodes import (
+    default_node_multi,
+    define_py_node,
+    shutdown_on_error,
+)
+from cudf_polars.experimental.rapidsmpf.utils import (
+    ChannelManager,
+    process_children,
+)
+from cudf_polars.experimental.utils import _concat
+
+if TYPE_CHECKING:
+    from rapidsmpf.streaming.core.context import Context
+
+    from cudf_polars.dsl.ir import IR, IRExecutionContext
+    from cudf_polars.experimental.rapidsmpf.core import SubNetGenerator
+    from cudf_polars.experimental.rapidsmpf.utils import ChannelPair
+
+
+async def get_small_table(
+    context: Context,
+    small_child: IR,
+    ch_small: ChannelPair,
+) -> list[DataFrame]:
+    """
+    Get the small-table DataFrame partitions from the small-table ChannelPair.
+
+    Parameters
+    ----------
+    context
+        The rapidsmpf context.
+    small_child
+        The small-table child IR node.
+    ch_small
+        The small-table ChannelPair.
+
+    Returns
+    -------
+    list[DataFrame]
+        The small-table DataFrame partitions.
+    """
+    small_chunks = []
+    while (msg := await ch_small.data.recv(context)) is not None:
+        small_chunks.append(TableChunk.from_message(msg))
+
+    if len(small_chunks) == 0:
+        raise ValueError("Empty small side")
```

Comment:
Is this possible, and if so under what conditions? It'd be nice to eliminate this case earlier if possible, or gracefully handle the case where you get an empty table here.